parachains/integration-tests/statemine/xcm/3_hrmp-open-channels.yml

```diff
@@ -161,248 +74,79 @@ tests:
-                    ]
-
-      - name: hrmp.hrmpInitOpenChannel (Assets Parachain → Penpal Parachain)
+      - name: hrmp.hrmpInitOpenChannel (Assets Parachain → PenPal Parachain)
         its:
-          - name: Assets Parchain sends a request to the Relay Chain to open a channel with a Penpal Parachain
+          - name: Open Assets Parachain to PenPal Parachain
             actions:
               - extrinsics:
                 - chain: *relay_chain
                   signer: *rc_signer
                   sudo: true
-                  pallet: xcmPallet
-                  call: send
+                  pallet: hrmp
+                  call: forceOpenHrmpChannel
                   args: [
-                    *ap_dest, # destination
-                    {
-                      v2: [ #message
-                        {
-                          Transact: {
-                            originType: Superuser,
-                            requireWeightAtMost: *weight_at_most,
-                            call: $xcm_init_open_channel_with_cp
-                          }
-                        }
-                      ]
-                    }
+                    1000,
+                    2000,
+                    8,
+                    8192
                   ]
                   events:
                     - name: sudo.Sudid
-                      attribute:
-                        type: Result<Null, SpRuntimeDispatchError>
-                        value: Ok
-                    - name: xcmPallet.Sent
-                    - name: dmpQueue.ExecutedDownward
-                      chain: *assets_parachain
-                      attribute:
-                        type: XcmV2TraitsOutcome
-                        isComplete: true
-                        threshold: [10, 10]
-                        value: 1,021,258,000
-                    - name: polkadotXcm.Sent
-                      chain: *assets_parachain
-                    - name: ump.ExecutedUpward
-                      timeout: 40000
-                      attribute:
-                        type: XcmV2TraitsOutcome
-                        isComplete: true
-                        threshold: [10, 10]
-                        value: 1,160,892,000
-                    - name: hrmp.OpenChannelRequested
-                      timeout: 40000
+                      attributes:
+                        - type: Result<Null, SpRuntimeDispatchError>
+                          value: Ok
+                    - name: hrmp.HrmpChannelForceOpened
               - queries:
-                  requested_channels:
+                  requested_channels_a_to_p:
                     chain: *relay_chain
                     pallet: hrmp
                     call: hrmpOpenChannelRequestsList
                     args: []
               - asserts:
                   equal:
                     args: [
-                      $requested_channels,
+                      $requested_channels_a_to_p,
```

Comment:
This assertion always fails for me. And it totally depends on the sequence of execution, if we change those tests around - it still fails. Because the list is empty. Since request processing happens on session start, this is going to be flaky even for the first test, depending on the machine and other variables. I suggest calling `force_process_hrmp_open` and then querying `HrmpIngressChannelsIndex` or `HrmpEgressChannelsIndex`. Otherwise we could just remove the assert completely as  the receipt of `HrmpChannelForceOpened` has to be enough.

Reply:
Agreed - as long as we have the event I think we're good. The tests in the next file would fail if the channel was not open.